### ISSUE TITLE
Search function will not return a SearchResult

### ISSUE THREAD

**Issue by iliion** [CS-SI/eodag#503]
I have configured an eodag instance having as a provider ONDA DIAS and also the respective username and password.
The objective is to search and download a product by its `ID` (The ID is `S2B_MSIL1C_20170717T153909_N0205_R011_T19TCJ_20170717T154916`)
I have written the following simple code in order to achieve the above.

**Code**
```py
...
    try:
        product: SearchResult= dag.search(id=product_id)
    except Exception as err:
        print(f"Failed to collect '{product_id}' product.", err)
...
```

The search function finds 1 result(s) on provider 'onda' but it doesnt return an _SearchResult_ successfully.

**Output**
```
python core/core.py --provider onda  --product_id S2B_MSIL1C_20170717T153909_N0205_R011_T19TCJ_20170717T154916
2022-09-12 19:13:05,923-15s eodag.config                     [INFO    ] (config           ) Loading user configuration from: /home/vagrant/Desktop/JRC_Repos/data-downloaders/eodag_download/eodag/core/config/eodag.yml
2022-09-12 19:13:06,356-15s eodag.core                       [DEBUG   ] (core             ) Opening product types index in /home/vagrant/.config/eodag/.index
2022-09-12 19:13:06,383-15s eodag.core                       [INFO    ] (core             ) Locations configuration loaded from /home/vagrant/Desktop/JRC_Repos/data-downloaders/eodag_download/eodag/core/config/locations.yml
2022-09-12 19:13:08,108-15s eodag.core                       [INFO    ] (core             ) Searching product with id 'S2B_MSIL1C_20170717T153909_N0205_R011_T19TCJ_20170717T154916' on provider: onda
2022-09-12 19:13:08,108-15s eodag.core                       [DEBUG   ] (core             ) Using plugin class for search: ODataV4Search
2022-09-12 19:13:08,109-15s eodag.plugins.search.qssearch    [DEBUG   ] (qssearch         ) Getting genric provider product type definition parameters for None
2022-09-12 19:13:08,109-15s eodag.plugins.search.qssearch    [DEBUG   ] (qssearch         ) Getting genric provider product type definition parameters for None
2022-09-12 19:13:08,110-15s eodag.plugins.search.qssearch    [DEBUG   ] (qssearch         ) Building the query string that will be used for search
2022-09-12 19:13:08,110-15s eodag.plugins.search.qssearch    [DEBUG   ] (qssearch         ) Retrieving queryable metadata from metadata_mapping
2022-09-12 19:13:08,111-15s eodag.plugins.search.qssearch    [INFO    ] (qssearch         ) Sending search request: https://catalogue.onda-dias.eu/dias-catalogue/Products?$format=json&$search=%22S2B_MSIL1C_20170717T153909_N0205_R011_T19TCJ_20170717T154916%22
2022-09-12 19:13:08,819-15s eodag.plugins.search.qssearch    [DEBUG   ] (qssearch         ) Sending metadata request: https://catalogue.onda-dias.eu/dias-catalogue/Products(de250e38-6e13-44cc-ba07-fc5411658ab7)/Metadata
2022-09-12 19:13:09,169-15s eodag.plugins.search.qssearch    [DEBUG   ] (qssearch         ) Adapting 1 plugin results to eodag product representation
2022-09-12 19:13:09,251-15s eodag.core                       [INFO    ] (core             ) Found 1 result(s) on provider 'onda'
Failed to collect 'S2B_MSIL1C_20170717T153909_N0205_R011_T19TCJ_20170717T154916' product. 
```

**Environment:**
 - Python version: 3.6.9
 - EODAG version: 2.5.2

**Additional context**

The metadata request is successfull (in the browser)
```https://catalogue.onda-dias.eu/dias-catalogue/Products(de250e38-6e13-44cc-ba07-fc5411658ab7)/Metadata```

As well as the request for the product:
```https://catalogue.onda-dias.eu/dias-catalogue/Products?$format=json&$search=%22S2B_MSIL1C_20170717T153909_N0205_R011_T19TCJ_20170717T154916%22```

**Comment by sbrunato** [comment]
Hello @iliion ,

in your example, the error is not displayed, so I do not know what can raise the error....

However, I can suggest you to:
- check that `SearchResult` is imported (`from eodag import SearchResult`)
- even if your example won't fail, it stores a tuple `(search_result, total_count)` in the variable `product`. You'd better unpack `search_result` and `total_count` in two variables (`search()` method returns 2 variables, unlike `search_all()`:
```py
product_id = "S2B_MSIL1C_20170717T153909_N0205_R011_T19TCJ_20170717T154916"
try:
    search_result, total_count = dag.search(id=product_id)
except Exception as err:
    print(f"Failed to collect '{product_id}' product.", err)
print(search_result)

2022-09-14 17:38:24,996 eodag.core                       [INFO    ] (core             ) Searching product with id 'S2B_MSIL1C_20170717T153909_N0205_R011_T19TCJ_20170717T154916' on provider: onda
2022-09-14 17:38:25,005 eodag.plugins.search.qssearch    [INFO    ] (qssearch         ) Sending search request: https://catalogue.onda-dias.eu/dias-catalogue/Products?$format=json&$search=%22S2B_MSIL1C_20170717T153909_N0205_R011_T19TCJ_20170717T154916%22
2022-09-14 17:38:26,002 eodag.plugins.search.qssearch    [DEBUG   ] (qssearch         ) Sending metadata request: https://catalogue.onda-dias.eu/dias-catalogue/Products(de250e38-6e13-44cc-ba07-fc5411658ab7)/Metadata
2022-09-14 17:38:26,476 eodag.plugins.search.qssearch    [DEBUG   ] (qssearch         ) Adapting 1 plugin results to eodag product representation
2022-09-14 17:38:26,587 eodag.core                       [INFO    ] (core             ) Found 1 result(s) on provider 'onda'
[EOProduct(id=S2B_MSIL1C_20170717T153909_N0205_R011_T19TCJ_20170717T154916, provider=onda)]

```
I also removed type annotation as it is [not supported for tuple unpacking](https://peps.python.org/pep-0526/#rejected-postponed-proposals)

**Comment by iliion** [comment]
Hello @sbrunato 
SearchResult was already imported correctly. --> `from eodag import SearchResult`

Nevertheless I copied your code and I get the same result. It is also true that no error is raised. It just fails to collect the Product.
Im quite puzzled.

For the record I already have another function implemented like this:
`products: SearchResult = dag.serch_all(**search_criteria, items_per_page=100)`

The latter is successfull

**Comment by sbrunato** [comment]
can you try this please ?
```py
import traceback

product_id = "S2B_MSIL1C_20170717T153909_N0205_R011_T19TCJ_20170717T154916"
try:
    search_result, total_count = dag.search(id=product_id)
    print(search_result)
except Exception:
    print(traceback.format_exc())
```
Then post the output. Thanks

**Comment by iliion** [comment]
I get this error 
```J_20170717T154916
2022-09-20 08:27:35,380-15s eodag.config                     [INFO    ] Loading user configuration from: /home/vagrant/Desktop/JRC_Repos/data-downloaders/eodag_download/eodag/core/config/eodag.yml
2022-09-20 08:27:35,890-15s eodag.core                       [INFO    ] Locations configuration loaded from /home/vagrant/Desktop/JRC_Repos/data-downloaders/eodag_download/eodag/core/config/locations.yml
2022-09-20 08:27:37,761-15s eodag.core                       [INFO    ] Searching product with id 'S2B_MSIL1C_20170717T153909_N0205_R011_T19TCJ_20170717T154916' on provider: onda
2022-09-20 08:27:37,764-15s eodag.plugins.search.qssearch    [INFO    ] Sending search request: https://catalogue.onda-dias.eu/dias-catalogue/Products?$format=json&$search=%22S2B_MSIL1C_20170717T153909_N0205_R011_T19TCJ_20170717T154916%22
2022-09-20 08:27:45,543-15s eodag.core                       [INFO    ] Found 1 result(s) on provider 'onda'
Traceback (most recent call last):
  File "core/core.py", line 49, in search_by_id
    product, total_count = dag.search(id=product_id)
  File "/home/vagrant/.local/lib/python3.6/site-packages/eodag/api/core.py", line 591, in search
    return self._search_by_id(search_kwargs.pop("id"), **search_kwargs)
  File "/home/vagrant/.local/lib/python3.6/site-packages/eodag/api/core.py", line 853, in _search_by_id
    guesses = self.guess_product_type(**results[0].properties)
  File "/home/vagrant/.local/lib/python3.6/site-packages/eodag/api/core.py", line 467, in guess_product_type
    raise NoMatchingProductType()
eodag.utils.exceptions.NoMatchingProductType
```

* Thanks for the `traceback`. Very useful

**Comment by sbrunato** [comment]
You're welcome.
Also, you should update eodag, as this issue has bee fixed in #380 ([v2.4.0](https://github.com/CS-SI/eodag/releases/tag/v2.4.0) release).
Or if you cannot update, specify the product type while searching: `dag.search(id=product_id, productType="S2_MSI_L1C")`

**Comment by iliion** [comment]
Ok that solved it. Thaks for your support. 
Cheers! :)
